rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java

```diff
@@ -214,6 +214,37 @@ public void setProposedFederation(Federation proposedFederation) {
         isProposedFederationSet = true;
     }
 
+    @Override
+    public Optional<Federation> getProposedFederation(FederationConstants federationConstants, ActivationConfig.ForBlock activations) {
+        if (!activations.isActive(RSKIP419)) {
+            return Optional.empty();
+        }
+
+        if (proposedFederation != null) {
+            return Optional.of(proposedFederation);
+        }
+
+        // reaching this point means the proposed federation was set to null
+        if (isProposedFederationSet) {
+            return Optional.empty();
+        }
```

Comment:
Although correct,  this got me confused and forced me to think about it for a while. I think we need to make it simpler and more straightforward

```suggestion
        if (proposedFederation != null || isProposedFederationSet) {
            return Optional.ofNullable(proposedFederation);
        }
```
This way it's consistent with getOldFederation for example.

Reply:
yeah at first i wrote it that way, but i thought it was better to actually know when it returns empty and when it returns an actual value. But no worries, i can change it